pkg/node/manager/node_manager.go

```diff
@@ -86,10 +87,13 @@ func (n *NodeManager) Register(ctx context.Context, request requests.RegisterReq
 		}, nil
 	}
 
-	request.Info.Approval = n.defaultApprovalState
-
-	if err := n.nodeInfo.Add(ctx, request.Info); err != nil {
-		return nil, errors.Wrap(err, "failed to save nodeinfo during node registration")
+	if err := n.store.Add(ctx, models.NodeState{
+		Info:     request.Info,
+		Approval: n.defaultApprovalState,
+		// NB(forrest): by virtue of a compute node calling this endpoint we can consider it connected
```

Comment:
This _should_ be okay, but using other signals than the heartbeat might lead to flapping, so if that happens, this might be a reasonable place to look.